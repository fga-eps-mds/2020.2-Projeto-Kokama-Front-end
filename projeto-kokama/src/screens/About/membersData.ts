import photoAndre from "../../assets/img/team-UnB/Andre.png";
import photoLeonardo from "../../assets/img/team-UnB/Leonardo.png";
import photoLieverton from "../../assets/img/team-UnB/Lieverton.png";
import photoWelison from "../../assets/img/team-UnB/Welison.jpeg";
import photoAna from "../../assets/img/team-UnB/Ana.png";
import photoFernando from "../../assets/img/team-UnB/Fernando.png";
import photoLais from "../../assets/img/team-UnB/Lais.png";
import photoLucas from "../../assets/img/team-UnB/Lucas.png";
import photoLuis from "../../assets/img/team-UnB/Luis.png";
import photoLuiz from "../../assets/img/team-UnB/Luiz.png";
import photoCelia from "../../assets/img/team-traduction/Celia.png";
import photoEraldo from "../../assets/img/team-traduction/Eraldo.png";
import photoMaria from "../../assets/img/team-traduction/Maria.png";
import photoMariana from "../../assets/img/team-traduction/Mariana.png";
import photoMatheus from "../../assets/img/team-traduction/Matheus.png";

export const membersEPS = [
  { name: "André Pinto", photo: photoAndre },
  { name: "Leonardo Medeiros", photo: photoLeonardo },
  { name: "Lieverton Santos", photo: photoLieverton },
  { name: "Welison Almeida", photo: photoWelison },
];

export const membersMDS = [
  { name: "Ana Júlia Briceño", photo: photoAna },
  { name: "Fernando Vargas", photo: photoFernando },
  { name: "Laís Portela", photo: photoLais },
  { name: "Lucas Monteiro", photo: photoLucas },
  { name: "Luís Gaboardi", photo: photoLuis },
  { name: "Luiz Campos", photo: photoLuiz },
];

export const membersTranslation = [
  { name: "Célia Rossi", photo: photoCelia },
  { name: "Eraldo Menezes", photo: photoEraldo },
  { name: "Maria Assis", photo: photoMaria },
  { name: "Mariana Ferreira", photo: photoMariana },
  { name: "Matheus José", photo: photoMatheus },
];