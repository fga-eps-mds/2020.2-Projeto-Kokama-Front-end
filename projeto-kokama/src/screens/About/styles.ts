import { StyleSheet, Dimensions } from "react-native";
import Colors from "../../assets/Colors";

// Constants and Global Variables
const window = Dimensions.get("window");
const screen = Dimensions.get("screen");



const styles = StyleSheet.create({
    container: {
        maxWidth: screen.width,
        height: window.height,
        backgroundColor: Colors.LIGHT_GRAY,
    },
    texts: {
        fontSize: 18,
        alignSelf: 'center',
    },
    EPS: {
        display: 'flex',
        maxHeight: 220,
        justifyContent: "space-between",
        flexDirection: 'row',
        flexWrap: 'wrap',
        borderWidth: 10,
        borderColor: Colors.LIGHT_GRAY,
    },
    MDS: {
        display: 'flex',
        maxHeight: 250,
        justifyContent: "space-between",
        flexDirection: 'row',
        flexWrap: 'wrap',
        borderWidth: 10,
        borderColor: Colors.LIGHT_GRAY,
    },
    photo: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
    },
    photos: {
        maxHeight: 80,
        maxWidth: 80,
        borderRadius: 250,
    },
    UnB: {
        display: 'flex',
        maxHeight: 100,
        maxWidth: 100,
    },
    NovaCartografia: {
        display: 'flex',
        maxHeight: 100,
        maxWidth: 200,
    },
});

export default styles;

