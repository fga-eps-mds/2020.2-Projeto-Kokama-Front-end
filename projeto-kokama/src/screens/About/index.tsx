import { View, Text, Image, ScrollView, SafeAreaView, TouchableWithoutFeedback, Linking } from "react-native";
import React, { useCallback } from "react";
import aboutStyle from "./styles";
import UserPhoto from "../../components/MemberPhoto";
import { teachers, membersEPS, membersMDS, membersTranslation } from "./membersData";
import Colors from "../../assets/Colors";

export default function About() {

  const wikiURL = "https://fga-eps-mds.github.io/2020.2-Projeto-Kokama-Wiki/"
  const redirectTo = useCallback(async () => {
    await Linking.openURL(wikiURL);
  }, []);

  return (
    <SafeAreaView style={aboutStyle.container}>
      <ScrollView>
        <Text style={[aboutStyle.title, {fontSize: 25}]}>Sobre:</Text>
        <View style={aboutStyle.about_view}>
          <Text style={aboutStyle.about_text}>
            O novo aplicativo Kokama é resultado das parcerias: do movimento de vitalização da língua Kokama,
            coordenado por Altaci Corrêa Rubim, Kokama; da universidade de Brasília-UnB, por meio do Campus Gama
            com os estudantes do curso Engenharia de Software; do campus Darcy Ribeiro, com os estudantes do curso
            de licenciatura de Português do Brasil como Segunda Língua-PBSL; da Universidade do Estado do Amazonas-UEA,
            por meio do Projeto Nova Cartografia Social da Amazônia e demais colaboradores.
          </Text>
          <Text style={aboutStyle.about_text}>
            O App Kokama Kinkin-APKKK tem como referência: o dicionário Kukama-Kukamiria, elaborado por Rosa Vallejos e
            Rosa Amias, do Peru, além de dados coletados nas oficinas de ensino e aprendizagem da língua Kokama.
          </Text>
          <Text style={aboutStyle.about_text}>
            O APP Kokama Kinkin é um aplicativo para smartphone que contribuirá com a política de vitalização da língua
            Kokama. Apresenta vocabulário contextualizado, atividades pedagógicas, além de possibilitar a
            atualização de dados pelos próprios Kokama.
          </Text>
          <Text style={aboutStyle.about_text}>
            Nossos mais sinceros agradecimentos ao prof. Hilmer Rodrigues Neri (UnB), ao prof. Alfredo Wagner Berno
            de Almeida (UEA) e, também, aos estudantes e colaboradores do projeto.
          </Text>
          <TouchableWithoutFeedback onPress={redirectTo}>
            <Text style={aboutStyle.about_text}>
              Para saber mais, acesse a <Text style={{ color: Colors.RED }}>página do projeto</Text>.
            </Text>
          </TouchableWithoutFeedback>
        </View>

        <Text style={aboutStyle.title}>Professores:</Text>

        <View style={aboutStyle.photoSection}>
          {teachers.map((teacher) => (
            <UserPhoto
              key={teacher.name}
              name={teacher.name}
              photo={teacher.photo}
            />
          ))}
        </View>

        <Text style={aboutStyle.title}>Equipe de Tradução:</Text>

        <View style={aboutStyle.photoSection}>
          {membersTranslation.map((memberTranslation) => (
            <UserPhoto
              key={memberTranslation.name}
              name={memberTranslation.name}
              photo={memberTranslation.photo}
            />
          ))}
        </View>

        <Text style={aboutStyle.title}>Equipe de Gerência - UnB:</Text>
        <View style={aboutStyle.photoSection}>
          {membersEPS.map((memberEPS) => (
            <UserPhoto
              key={memberEPS.name}
              name={memberEPS.name}
              photo={memberEPS.photo}
            />
          ))}
        </View>

        <Text style={aboutStyle.title}>Equipe de Desenvolvimento - UnB:</Text>
        <View style={aboutStyle.photoSection}>
          {membersMDS.map((memberMDS) => (
            <UserPhoto
              key={memberMDS.name}
              name={memberMDS.name}
              photo={memberMDS.photo}
            />
          ))}
        </View>

        <Text style={aboutStyle.title}>Apoiadores</Text>
        <View style={aboutStyle.photoSection}>
          <Image
            source={require("../../assets/img/UnB-Logo.png")}
            style={aboutStyle.UnB}
          />
          <Image
            source={require("../../assets/img/Nova-Cartografia.png")}
            style={aboutStyle.NovaCartografia}
          />
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}
