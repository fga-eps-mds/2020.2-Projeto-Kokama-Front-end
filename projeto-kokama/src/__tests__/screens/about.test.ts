import renderer from 'react-test-renderer';
import About from '../../screens/About';

test("Check About screen", () => {
    
    let about = About()
    let tree = renderer.create(about)
    expect(tree).toMatchSnapshot();
})