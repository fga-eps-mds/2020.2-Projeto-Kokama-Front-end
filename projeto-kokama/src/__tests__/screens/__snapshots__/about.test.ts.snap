// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check About screen 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FAFAFA",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 20,
            "marginVertical": 15,
          }
        }
      >
        Sobre:
      </Text>
      <View
        style={
          Object {
            "justifyContent": "center",
            "marginBottom": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "lineHeight": 20,
              "paddingHorizontal": 40,
              "textAlign": "justify",
            }
          }
        >
          O novo aplicativo Kokama é resultado das parcerias: do movimento de vitalização da língua Kokama, coordenado por Altaci Corrêa Rubim, Kokama; da universidade de Brasília-UnB, por meio do Campus Gama com os estudantes do curso Engenharia de Software; do campus Darcy Ribeiro, com os estudantes do curso de licenciatura de Português do Brasil como Segunda Língua-PBSL; da Universidade do Estado do Amazonas-UEA, por meio do Projeto Nova Cartografia Social da Amazônia e demais colaboradores.
        </Text>
        <Text
          style={
            Object {
              "lineHeight": 20,
              "paddingHorizontal": 40,
              "textAlign": "justify",
            }
          }
        >
          O App Kokama Kinkin-APKKK tem como referência: o dicionário Kukama-Kukamiria, elaborado por Rosa Vallejos e Rosa Amias, do Peru, além de dados coletados nas oficinas de ensino e aprendizagem da língua Kokama.
        </Text>
        <Text
          style={
            Object {
              "lineHeight": 20,
              "paddingHorizontal": 40,
              "textAlign": "justify",
            }
          }
        >
          O APP Kokama Kinkin é um aplicativo para smartphone que contribuirá com a política de vitalização da língua Kokama. Apresenta vocabulário contextualizado, áudio, atividades pedagógicas, além de possibilitar a atualização de dados pelos próprios Kokama.
        </Text>
        <Text
          style={
            Object {
              "lineHeight": 20,
              "paddingHorizontal": 40,
              "textAlign": "justify",
            }
          }
        >
          Nossos mais sinceros agradecimentos ao prof. Hilmer Rodrigues Neri (UnB), ao prof. Alfredo Wagner Berno de Almeida (UEA) e, também, aos estudantes e colaboradores do projeto.
        </Text>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 20,
            "marginVertical": 15,
          }
        }
      >
        Equipe de Tradução:
      </Text>
      <View
        style={
          Object {
            "borderColor": "#FAFAFA",
            "borderWidth": 10,
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 20,
            "marginVertical": 15,
          }
        }
      >
        Equipe de Gerência-UnB:
      </Text>
      <View
        style={
          Object {
            "borderColor": "#FAFAFA",
            "borderWidth": 10,
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 20,
            "marginVertical": 15,
          }
        }
      >
        Equipe de Desenvolvimento-UnB:
      </Text>
      <View
        style={
          Object {
            "borderColor": "#FAFAFA",
            "borderWidth": 10,
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": 100,
              "margin": 8,
            }
          }
        >
          <Image
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 250,
                "maxHeight": 80,
                "maxWidth": 80,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 20,
            "marginVertical": 15,
          }
        }
      >
        Apoiadores
      </Text>
      <View
        style={
          Object {
            "borderColor": "#FAFAFA",
            "borderWidth": 10,
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "display": "flex",
              "maxHeight": 100,
              "maxWidth": 100,
            }
          }
        />
        <Image
          source={1}
          style={
            Object {
              "display": "flex",
              "maxHeight": 100,
              "maxWidth": 200,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
